spring-kafka/spring-kafka/src/main/java/org/springframework/kafka/core/DefaultKafkaProducerFactory.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.kafka.core;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.kafka.clients.producer.KafkaProducer;
+import org.apache.kafka.clients.producer.Producer;
+
+/**
+ * @author Gary Russell
+ *
+ */
+public class DefaultKafkaProducerFactory<K, V> implements ProducerFactory<K, V> {
+
+	private final Map<String, Object> configs;
+
+	public DefaultKafkaProducerFactory(Map<String, Object> configs) {
+		this.configs = new HashMap<>(configs);
+	}
+
+	@Override
+	public Producer<K, V> createProducer() {
+		return new KafkaProducer<>(this.configs);
```

Comment:
Since the producer is thread-safe, we should probably keep serving up the same instance until it's closed. Since it doesn't have a `isOpen()` method, though, it means we'd have to wrap it.